sklearn/linear_model/cd_fast.pyx

```diff
@@ -201,26 +240,25 @@ def enet_coordinate_descent(np.ndarray[DOUBLE, ndim=1] w,
 
                 if w_ii != 0.0:
                     # R += w_ii * X[:,ii]
-                    daxpy(n_samples, w_ii,
-                          <DOUBLE*>(X.data + ii * n_samples * sizeof(DOUBLE)),
-                          1, <DOUBLE*>R.data, 1)
+                    axpy(n_samples, w_ii, &X_data[ii * n_samples], 1,
+                         R_data, 1)
 
                 # tmp = (X[:,ii]*R).sum()
-                tmp = ddot(n_samples,
-                           <DOUBLE*>(X.data + ii * n_samples * sizeof(DOUBLE)),
-                           1, <DOUBLE*>R.data, 1)
+                if floating is float:
+                    tmp = dsdot(n_samples, &X_data[ii * n_samples], 1, R_data, 1)
+                else:
+                    tmp = dot(n_samples, &X_data[ii * n_samples], 1, R_data, 1)
 
                 if positive and tmp < 0:
                     w[ii] = 0.0
                 else:
                     w[ii] = (fsign(tmp) * fmax(fabs(tmp) - alpha, 0)
-                             / (norm_cols_X[ii] + beta))
+                            / (norm_cols_X[ii] + beta))
```

Comment:
Doctest is someone else's problem:
https://github.com/scikit-learn/scikit-learn/pull/7083/files#r72380271

On 27 July 2016 at 14:40, Yen <notifications@github.com> wrote:

> In sklearn/linear_model/cd_fast.pyx
> <https://github.com/scikit-learn/scikit-learn/pull/6913#discussion_r72380018>
> :
>
> >
> >                  if positive and tmp < 0:
> >                      w[ii] = 0.0
> >                  else:
> >                      w[ii] = (fsign(tmp) * fmax(fabs(tmp) - alpha, 0)
> > -                             / (norm_cols_X[ii] + beta))
> > +                            / (norm_cols_X[ii] + beta))
>
> I've fixed this, but it seems that some doctest is now failing.
> Should I rebase master and force push the branch?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/6913/files/dfaedcfa6374890580479e0cc9f927806af95179#r72380018>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz68CUpaNnaAxcmVP1BPghl8YwH4egks5qZuFPgaJpZM4I6SId>
> .
>